src/main/java/online/book/store/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +39,17 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, UpdateBookRequestDto bookRequestDto) {
```

Comment:
```suggestion
    @Override
    @Transactional
    public BookDto updateById(Long id, UpdateBookRequestDto bookRequestDto) {
```

we make more than one call to db, so it has to be isolated